src/v/kafka/server/handlers/describe_configs.cc

```diff
@@ -77,12 +79,67 @@ static ss::sstring describe_as_string(const T& t) {
     return ssx::sformat("{}", t);
 }
 
+// property_config_type maps the datatype for a config property to
+// describe_configs_type. Currently class_type and password are not used in
+// Redpanda so we do not include checks for those types. You may find a similar
+// mapping in Apache Kafka at
+// https://github.com/apache/kafka/blob/be032735b39360df1a6de1a7feea8b4336e5bcc0/core/src/main/scala/kafka/server/ConfigHelper.scala
+// Type sizes are based on Java and taken from
+// https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html
+template<typename T>
+consteval describe_configs_type property_config_type() {
+    if constexpr (reflection::is_std_optional<T>) {
+        return property_config_type<typename T::value_type>();
+    } else if constexpr (reflection::is_tristate<T>) {
+        return property_config_type<typename T::value_type>();
+    } else if constexpr (std::is_same_v<T, bool>) {
+        return describe_configs_type::boolean;
+    } else if constexpr (std::is_same_v<T, ss::sstring>) {
+        return describe_configs_type::string;
+    } else if constexpr (std::is_integral_v<T>) {
+        if (std::is_same_v<T, short> || std::is_same_v<T, ushort>) {
+            return describe_configs_type::short_type;
+        } else if (std::is_same_v<T, int> || std::is_same_v<T, uint>) {
```

Comment:
what about int vs int16_t vs int32_t? Same for uint, short and long. Maybe there is a better type trait or you can make your own.

Reply:
I tested alot of the int variants and they all map to one of these types.
`int` -> already covered here
`int16_t` -> `int`
`int32_t` -> `int`
`int64_t` -> `long`
`size_t` -> `long`

> Maybe there is a better type trait or you can make your own.

I don't think we'll be able to cover all possible types; however, all the types from our configs are covered here. In a future where we add a new config, we will have to rely on the developer to either use already-supported-types or update this mapping. 
To support this point, we have the static_assert check so the developer will immediately know if their using an unsupported type.

Reply:
> I tested several of the int variants and they all map to one of these types. `int` -> already covered here `int16_t` -> `int` `int32_t` -> `int` `int64_t` -> `long` `size_t` -> `long`

These are system dependent, but again, the static_assert check will tell us if we missed something no?

Reply:
> > To support this point, we have the static_assert check
> 
> Note that the `static_assert` at the end of this function does not catch unsupported integer types (e.g. `signed char`).



> > > I tested several of the int variants and they all map to one of these types. `int` -> already covered here `int16_t` -> `int` `int32_t` -> `int` `int64_t` -> `long` `size_t` -> `long`
> > 
> > 
> > These are system dependent, but again, the static_assert check will tell us if we missed something no?
> 
> nit: `long` is 32 bit in MSVC, so we can run into a sudden surprise here should ever this code be destined for Windows. A minor concern though (that will probably never happen given our attachment to XFS), but in case `desctibe_config_types` defines a network protocol type (apparently it does), using fixed-bits types like `(u)int??_t` seems to be a cleaner approach.
> 
> [FYR](https://pvs-studio.com/en/blog/terms/0012/)

I'll add in the `(u)intXX_t` types and I'll also look back through C++ fundamental types to account for the ones not represented yet.

Reply:
Done

Reply:
Oh I didn't account for MSVC. I'll put in a compiler macro in the morning.

Reply:
> Oh I didn't account for MSVC. I'll put in a compiler macro in the morning.

Spoke with the team in slack. Windows development is not a concern for now so we can ignore.